src/granite/fields.cr

```diff
@@ -14,6 +14,7 @@ module Granite::Fields
 
   # specify the fields you want to define and types
   macro field(decl, **options)
+    {% raise "The type of #{@type.name}##{decl.var} cannot be a Union.  The 'field' macro declares the type as nilable by default.  Use the 'field!' macro to declare a not nilable field." if decl.type.is_a? Union %}
```

Comment:
I love these error messages!

They're long enough that the suffix `if` kind of gets lost, would it be better to break this out into multiple lines for maintainability?

Reply:
:man_shrugging: I don't it really matters.

Reply:
@robacarp are you ok with this?  we can refactor this later if you feel it's hard to read.

Reply:
@amberframework/contributors Any other feedback?  If not, I will merge.